woocommerce/woocommerce-ios
# REST API: Migrate coupon endpoints to use REST API

Closes: #8390 



## Description

This PR added changes to coupon endpoints to call REST requests with Jetpack requests as fallback. Changes include:
- Updated `CouponListMapper` and `CouponMapper` to parse coupons regardless of whether the `data` envelope is present in the response.
- Updated the `RESTRequest` with an initializer that accepts a site URL and a Jetpack request.
- Updated `CouponsRemote`, `CouponAction`, and `CouponStore` to accept site URL as input.
- Updated the UI layer following changes in the Yosemite layer.

## Testing instructions


- Navigate to Menu > Coupons. If the option is unavailable, navigate to Settings > Experimental Features > Enable coupon management.
- Notice that the coupon list is loaded properly.
- Try searching, creating, editing, deleting coupons. Everything should work as before. 

## Screenshots

N/A

---
